src/auth/driveAuthenticator.ts

```diff
@@ -1,18 +1,18 @@
 import { env, Uri, window, commands } from "vscode";
 import { CredentialsManager, CREDENTIALS_JSON_SERVICE, TOKENS_JSON_SERVICE } from "../drive/credentials/credentialsManager";
 const { OAuth2Client } = require('google-auth-library');
-const { google } = require('googleapis');
 const http = require('http');
 const url = require('url');
 const destroyer = require('server-destroy');
 import * as fs from "fs";
 import { CONFIGURE_CREDENTIALS_COMMAND } from "../extension";
+import { DriveExtensionSettings } from "../driveExtensionSettings";
 
 // If modifying these scopes, delete token.json.
 const SCOPES = ['https://www.googleapis.com/auth/drive.file', 'https://www.googleapis.com/auth/drive', 'https://www.googleapis.com/auth/drive.readonly'];
 
 // Port and endpoint where this extension will listen for the OAuth2 token
-const OAUTH_PORT = 3000;
+const OAUTH_PORT = new DriveExtensionSettings().getAuthPort();
```

Comment:
I am curious about what would happen if we manually open the JSON settings file and define a text as the value for this setting. For example, defining _"foo"_ for this new setting _google.drive.authPort_ instead of a number... would an error / exception be thrown here?

Reply:
The extension doesn't crash if we set the setting with a text value. File settings.json tells the user that the type of the parameter is incorrect. Is this sufficient?

Reply:
This is interesting. But I mean if we manually set a text value for this setting via an external editor, what would happen [here](https://github.com/GustavoASC/google-drive-vscode/blob/3ed2173bd74a94f4f35a4daaf510d0b22b0113de/src/auth/driveAuthenticator.ts#L215) when the server tries to listen on the giver port with invalid value?

```
      // Effectively listens on the HTTP port until OAuth2 sends a request to it
      server.listen(OAUTH_PORT, async () => {
         ...
      }
```
`settings.json tells the user that the type of the parameter is incorrect`: yep, I think this is enough for our scenario, as setting manually via an external editor would be an edge case and there may be no need to address this now. But I'm still curious about what would happen in that aforementioned scenario.